src/content/blog/drupalcon-barcelona-2024.md

```diff
@@ -0,0 +1,64 @@
+
+## Ideas from the Driesnote
+
+A DrupalCon tradition is the keynote by project founder and leader Dries Buytaert called the [Driesnote](https://dri.es/state-of-drupal-presentation-september-2024).
+
+There were many inspirational topics here, with people focusing on "Starshot" or "Drupal CMS", which seems to be coming along well.
+
+A key inspiration, though, was the idea of having organizations sponsor sections of the Drupal documentation. It's always been impossible for volunteers to keep up with the Drupal docs, and having a the professional organization Drupalize.me maintain sections of the doc formally is a great idea. The new idea is like the "Adopt a highway" initiative used for decades in the US, where private businesses or other organizations pay for highway cleanup. For $2400/year, an organization can now [sponsor a section of the Drupal.org docs](https://drupalize.me/blog/adopt-document-sponsor-drupal-cms-documentation) to get it to the level it should be.
+
+I think DDEV could do this same thing, and our favorite agency, CMS, and hosting provider partners could sponsor DDEV features so they could be properly maintained over the long term.
+
+## Things I Learned
+
+* `ddev composer` doesn't work right when there are multiple `composer.json`. For most projects, there is one main `composer.json` which can be specified with DDEV's `composer_root` directive, but in projects set up with the [joachim-n/drupal-core-development-project](https://github.com/joachim-n/drupal-core-development-project) there is a main `composer.json` and then also the one in `repos/drupal`, which is usually the one that might change if you're working on Drupal core. In this situation, it's important to either change `composer_root` to `repos/drupal` or to `ddev ssh` and `cd repos/drupal` and do composer activities there.
+* When you need only to update the hash in the `composer.json` the technique is to use `composer update --lock`.
+* Some people using Visual Studio Code ("vs code") have been using the [Remote Explorer](https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-explorer) and [Devcontainers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extensions to work inside DDEV's web container instead of using the traditional DDEV approaches on the host. [Mike Anello](https://www.drupaleasy.com/) has been promoting this technique in his outstanding long-form training sessions, so I was super interested in trying it out. Instead of using `ddev xdebug on` you use `enable_xdebug`. Instead of `ddev exec phpcs` or `ddev exec phpstan` you directly use `phpcs` or `phpstan`. It's a fascinating approach, and has some strengths and some frictions compared to the traditional approach. (Note that the fantastic [Visual Studio Code DDEV Manager](https://marketplace.visualstudio.com/items?itemName=biati.ddev-manager) extension is not used at all here, and actually can break things.)
```

Comment:
```suggestion
* Some people using Visual Studio Code (VS Code) have been using the [Remote Explorer](https://marketplace.visualstudio.com/items?itemName=ms-vscode.remote-explorer) and [Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) extensions to work inside DDEV's web container instead of using the traditional DDEV approaches on the host. [Mike Anello](https://www.drupaleasy.com/) has been promoting this technique in his outstanding long-form training sessions, so I was super interested in trying it out. Instead of using `ddev xdebug on` you use `enable_xdebug`. Instead of `ddev exec phpcs` or `ddev exec phpstan` you directly use `phpcs` or `phpstan`. It's a fascinating approach, and has some strengths and some frictions compared to the traditional approach. (Note that the fantastic [Visual Studio Code DDEV Manager](https://marketplace.visualstudio.com/items?itemName=biati.ddev-manager) extension is not used at all here, and actually can break things.)
```

Reply:
And this one?

Reply:
Just lost is somehow :) thanks.